Mage/src/main/java/mage/abilities/common/BlocksOrBlockedAttachedTriggeredAbility.java

```diff
@@ -2,41 +2,65 @@
 
 import mage.abilities.TriggeredAbilityImpl;
 import mage.abilities.effects.Effect;
+import mage.constants.AttachmentType;
 import mage.constants.Zone;
 import mage.game.Game;
 import mage.game.events.GameEvent;
 import mage.game.permanent.Permanent;
 import mage.target.targetpointer.FixedTarget;
 
 /**
- * @author awjackson
+ * @author awjackson & TiagoMDG
  */
 public class BlocksOrBlockedAttachedTriggeredAbility extends TriggeredAbilityImpl {
 
-    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect) {
-        this(effect, false);
+    private final AttachmentType attachmentType;
+    private final boolean selfTarget;
+
+    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, AttachmentType attachmentType, boolean selfTarget) {
+        this(effect, attachmentType, selfTarget, false);
     }
-    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, boolean optional) {
+
+    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, AttachmentType attachmentType, boolean selfTarget, boolean optional) {
```

Comment:
Please add a comment to clarify the meaning of `boolean selfTarget` for future devs.